source/Halibut.Tests/Util/AsyncEx/TaskExtensionsFixture.cs

```diff
@@ -0,0 +1,117 @@
+using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using FluentAssertions;
+using Halibut.Util.AsyncEx;
+using NUnit.Framework;
+
+namespace Halibut.Tests.Util.AsyncEx
+{
+    [TestFixture]
+    public class TaskExtensionsFixture
+    {
+        [Test]
+        public async Task When_TaskCompletesWithinTimeout_TaskCompletesSuccessfully()
+        {
+            var triggered = false;
+            var task = Task.Run(async () =>
+            {
+                await Task.Delay(TimeSpan.FromMilliseconds(100));
+                triggered = true;
+            });
+            await task.TimeoutAfter(TimeSpan.FromMilliseconds(300), CancellationToken.None);
+            triggered.Should().Be(true, "the task should have triggered");
+        }
+        
+        [Test]
+        public async Task When_TaskDoesNotCompleteWithinTimeout_ThrowsTimeoutException()
+        {
+            var triggered = false;
+            var task = Task.Run(async () =>
+            {
+                await Task.Delay(TimeSpan.FromMilliseconds(200));
+                triggered = true;
+            });
+            Func<Task> act = async () => await task.TimeoutAfter(TimeSpan.FromMilliseconds(100), CancellationToken.None);
+            act.ShouldThrow<TimeoutException>();
+            triggered.Should().Be(false, "the task should have been aborted");
```

Comment:
There's a race condition here. The underlying task does not "get aborted" and this test presumably only passes because the assertion is executed before the underlying task sets `triggered` to `true`.